File: algorithms/1_easy/F/01991_find_the_middle_index_in_array/solution.c
int findMiddleIndex(const int *restrict nums, const int len) {
    int suff = 0;
    for (int i = 0; i < len; i += 1) { suff += nums[i]; }
    int pref = 0;;

    for (int i = 0; i < len; i += 1) {
        suff -= nums[i];

        if (suff == pref) {
            return i;
        }

        pref += nums[i];
    }

    return -1;
}

File: algorithms/1_easy/F/01991_find_the_middle_index_in_array/solution.rs
impl Solution {
    pub fn find_middle_index(nums: Vec<i32>) -> i32 {
        let mut suff: i32 = nums.iter().sum();
        let mut pref: i32 = 0;
        let len = nums.len();

        for i in 0..len {
            suff -= nums[i];

            if suff == pref {
                return i.try_into().unwrap();
            }

            pref += nums[i];
        }

        -1
    }
}
